Add checksmtp functionality for SMTP configuration verification

fer opened the pull request:
## Description

Introduces the `checksmtp` command to the Bitwarden self-hosted management script, enabling administrators to easily verify the SMTP configuration directly from the command line. This functionality aims to streamline the troubleshooting process of email delivery issues by providing a quick and straightforward method to test the SMTP server connection, authentication, and secure communication.

## Features 
- [x] Validates the connection to the SMTP server specified in the global.override.env file, ensuring that the Bitwarden instance can communicate with the mail server.
- [x] Confirms the validity of the SMTP username and password, ensuring that Bitwarden is capable of authenticating with the SMTP server for email delivery.
- [x] Verifies the secure communication with the SMTP server, based on the SSL configuration `globalSettings__mail__smtp__ssl` to ensure that emails are sent over a secure channel.

## Usage examples

To use this functionality, run the following command from the Bitwarden self-hosted script directory:

```bash
./bitwarden.sh checksmtp
```

When SMTP configuration is wrong:

```
SMTP authentication failed or connection error occurred.
```

When SMTP configuration is right:

```
SMTP settings are correct.
```

## Testing

Please review this PR for inclusion in the next release of Bitwarden self-hosted. Your feedback and suggestions are welcome!

vgrassia reviewed (commented):
I would check to see if the `openssl` binary is available before calling it. Shouldn't the `AUTH LOGIN` section only execute if there is a `username` or `password` defined?  This doesn't work on my self-host that doesn't use authentication. Also, I'm not sure every SMTP server returns `235` for a successful message, it could be any number of `2XX` codes.  How do we make sure it actually sent the message?

fer commented:
@vgrassia, thank you very much for your comments! 

I've implemented the following enhancements:

- Added a preliminary check to ensure `openssl` is installed before proceeding.
- Modified the script to accept any 2xx SMTP status code as a successful response.
- Modified the script to only attempt the `AUTH` section if both the username and password are provided and are not empty.

Currently, this script is designed to verify connectivity without sending any messages. However, I'm open to incorporating a feature to send a test message if we determine it's beneficial for our testing process. 

Please let me know if you'd like me to proceed with that addition as your valuable suggestions.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bitwarden/self-host?pullRequest=231) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bitwarden/self-host?pullRequest=231) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bitwarden/self-host?pullRequest=231) it.</sub>

vgrassia commented on bitwarden.sh:
```diff
@@ -190,6 +190,56 @@ function compressLogs() {
+    else
+        ssl_command="-starttls smtp"
+    fi
+
```
```suggestion

set +e
```
If the user has not changed the SMTP settings from the defaults on their installation, the `openssl` command will fail and since our script has `set -e` at the top, it will exit immediately without any output. This changes it so that it will at least print the error `SMTP authentication failed or connection error occurred.`

vgrassia commented on bitwarden.sh:
```diff
@@ -267,6 +317,10 @@ case $1 in
         compressLogs $OUTPUT $2 $3
         ;;
+    "checksmtp")
+        # checkOutputDirExists
```
```suggestion
        checkOutputDirExists
```

vgrassia commented on bitwarden.sh:
```diff
@@ -190,6 +190,57 @@ function compressLogs() {
+        ssl_command="-starttls smtp"
+    fi
+
+set +e
```
```suggestion
    set +e
```
Sorry, I forgot the indentation.
